### PR TITLE
fake aanderaa

### PR DESCRIPTION
**Pull request by russelldeguzman**
`cmake ../.. -DCMAKE_TOOLCHAIN_FILE=../../cmake/arm-none-eabi-gcc.cmake -DBSP=bm_mote_bristleback_v1_0 -DCMAKE_BUILD_TYPE=Debug -DCMAKE_APP_TYPE=bristleback -DAPP=aanderaa -DCMAKE_AANDERAA_TYPE=FAKE_AANDERAA`

### CODE REVIEW

File: src/apps/bridge/aanderaaController.cpp [hunk 1]
```diff
@@ -39,6 +39,9 @@ typedef struct Aanderaa {
 
 public:
   bool subscribe();
+#ifdef AANDERAA_FAKEOUT
```
Comment (towynlin) [inline src/apps/bridge/aanderaaController.cpp]: Looks like this need to change to FAKE_AANDERAA.

Comment (victorsowa12) [inline src/apps/bridge/aanderaaController.cpp]: Should this be `FAKE_AANDERAA`?

Comment (russelldeguzman) [reply src/apps/bridge/aanderaaController.cpp]: Shouldnt be there at all lolol
whoops!

File: src/apps/bristleback_apps/aanderaa/user_code/user_code.cpp [hunk 2]
```diff
@@ -409,4 +416,117 @@ void loop(void) {
...
     }
   }
+#endif // FAKE_AANDERAA
 }
+
+#ifdef FAKE_AANDERAA
+static void spoof_aanderaa() {
+  static constexpr uint32_t FAKE_STATS_PERIOD_MS = 60 * 1000;
+  static uint32_t fake_stats_timer_start = uptimeGetMs();
+  if (((u_int32_t)uptimeGetMs() - fake_stats_timer_start >= FAKE_STATS_PERIOD_MS)) {
```
Comment (towynlin) [inline src/apps/bristleback_apps/aanderaa/user_code/user_code.cpp]: Use `uint32_t` without the extra underscore.

**Review by victorsowa12 — APPROVED** [review]

File: src/apps/bristleback_apps/aanderaa/user_code/user_code.cpp [hunk 3]
```diff
@@ -409,4 +416,117 @@ void loop(void) {
...
+    d.direction_deg_m = rand() % 337;
+    d.east_cm_s = rand() % -19;
+    d.heading_deg_m = rand() % 153;
+    d.north_cm_s = rand() % 41;
+    d.ping_count = rand() % 100;
+    d.tilt_x_deg = rand() % 2;
+    d.tilt_y_deg = rand() % 20;
+    d.max_tilt_deg = rand() % 3;
+    d.std_tilt_deg = rand() % 39;
+    d.temperature_deg_c = rand() % 20;
```
Comment (towynlin) [inline src/apps/bristleback_apps/aanderaa/user_code/user_code.cpp]: Fine for now. We can continue tweaking later to be more realistic if it helps with testing. Just make sure it doesn't result in NAN for bad values.

**Review by towynlin — APPROVED** [review]

**Pull request merged**